### PR TITLE
Bugfix/ip 238 angular2 check

### PR DESCRIPTION
**Pull request by mobinni**
Concerning #238 I switched out some imports to require statements for conditional loading. You check if your window object has the **ng** attribute. 

If so you're probably running Angular2 so you can start your adapter and hook into the DOM. 

I don't feel it's the cleanest fix in the world, since I prefer ES6 imports, but for the sake of low coupling I felt that backend.ts should be responsible for only loading **IF** ```window.ng``` is defined.

### CODE REVIEW

File: src/backend/backend.ts [hunk 1]
```diff
@@ -14,10 +12,20 @@ let channel = {
...
+
+if (window.hasOwnProperty('ng')) {
+  const {Angular2Adapter} = require('./adapters/angular2');
+  const {DomController} = require('./controllers/dom');
+
+  adapter = new Angular2Adapter();
+  dom = new DomController(adapter, channel);
+}
+
+if(adapter && dom) {
```
Comment (sumitarora) [inline src/backend/backend.ts]: Missing whitespace

File: src/backend/backend.ts [hunk 2]
```diff
@@ -14,10 +12,20 @@ let channel = {
...
+  dom.hookIntoBackend();
+  adapter.setup();
+}
```
Comment (sumitarora) [inline src/backend/backend.ts]: Trailing Whitespace

**Comment by sumitarora** [comment]
@mobinni We can wrap second if inside the first one we don't need it separate other than that looks good. Also, Can you add a comment there why we are doing it like this?

**Comment by mobinni** [comment]
@sumitarora done!

**Comment by vanessayuenn** [comment]
Hey @mobinni, thanks a lot for the contribution! After reviewing your solution, I agree with your stance on preferring `import` over `require`. Here's what I have in mind in terms of a cleaner approach:
- Have `content-script` be responsible for determining whether `backend.js` should be injected.
- Create a separate script with a simple function to determine whether `ng` is available (similar to what you have already), if so, post a message for `content-script`.
- Inject said script to the page batarangle is trying to debug, just like how `backend.js` is injected.

This way _does_ require touching more parts of the application, so please DM me on slack if you have any question. :)

**Pull request closed without merging**